lambdas/mail_send.js

```diff
@@ -4,17 +4,47 @@ if (process.env.NODE_ENV === 'development') {
 
 const domain = process.env.MAILGUN_DOMAIN;
 const apiKey = process.env.MAILGUN_API_KEY;
+const secret_key = process.env.SECRET_SEND_KEY;
 
+const { send } = require('micro');
+const statuses = require('statuses');
+const { upload } = require('micro-upload')
 const mailgun = require('mailgun-js');
 
-const mg = mailgun({ apiKey, domain });
-const data = {
-  from: 'WebConf <no-reply@webconf.tech>',
-  to: 'gahs94@gmail.com',
-  subject: 'WebConf • Nuestro newsletter 🙈',
-  text: 'Hola perrix! :D'
-};
+module.exports = upload(async (req, res) => {
+  if (!(req.files && req.files.file && req.files.file.mimetype === 'text/html')) {
+    return send(res, statuses['bad request'], 'Error - No file uploaded or wrong file format');
+  }
```

Comment:
If you are doing _"early exit"_ with the file validation, why don't you do the same with the secret?

```js
if (!(req.files && req.files.file && req.files.file.mimetype === 'text/html')) {
  return ...
} else if (secret !== secret_key) {
  return ...
} 

...
```

Reply:
Agreed, but use a separate if. Since we're doing early exits, else if is unnecessary.